File: FocusLit-Server/src/main/java/com/focus/lit/model/Session.java
package com.focus.lit.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Table(name = "sessions")
public class Session {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "tag_id", referencedColumnName = "id")
    private Tag tag;

    @Column(name = "start_time")
    private LocalDateTime startTime;

    @Column(name = "work_duration")
    private int workDuration;

    @Column(name = "wait_duration")
    private int waitDuration;

    private int score;

    @Column(name = "score_multiplier")
    private float scoreMultiplier;

    @Column(name = "is_completed")
    private boolean isCompleted;
    
    @Column(name="completed_work_duration")
    private int completedWorkDuration;

}

File: FocusLit-Android/app/src/main/java/com/focus/lit/data/model/Session.kt
package com.focus.lit.data.model

import android.os.Build
import androidx.annotation.RequiresApi
import java.time.DayOfWeek
import java.time.LocalDateTime

data class Session(
    val userId: Int,
    val tagId: Int,
    val startTime: String,
    val workDuration: Long,          // in seconds or minutes
    val waitDuration: Long,
    val score: Int,
    val scoreMultiplier: Double,
    val completedWorkDuration: Long, // in same units as workDuration
    val isCompleted: Boolean
)
